### PR TITLE
Use 0x.js subscribeAsync

### PR DESCRIPTION
**Pull request by fabioberger**
This PR:
- Uses 0x.js's zeroEx.exchange.subscribeAsync for listening to OTC LogFill events.
- Stops watching for events when Blockchain class is destroyed.
- renames several variables for clarity.

### CODE REVIEW

File: ts/blockchain.ts [hunk 1]
```diff
@@ -47,7 +54,7 @@ export class Blockchain {
     private dispatcher: Dispatcher;
     private web3Wrapper: Web3Wrapper;
     private exchange: ContractInstance;
-    private exchangeLogFillEvents: any[];
+    private exchangeLogFillEventEmitters: any[];
```
Comment (LogvinovLeon) [inline ts/blockchain.ts]: `ContractEventEmitter`

File: ts/blockchain.ts [hunk 2]
```diff
@@ -343,19 +351,23 @@ export class Blockchain {
             // transactions where an account is either the `maker` or `taker`, we loop
             // through all fill events, and filter/cache them client-side.
             const filterIndexObj = {};
-            this.startListeningForExchangeLogFillEvents(filterIndexObj);
+            await this.startListeningForExchangeLogFillEventsAsync(filterIndexObj);
```
Comment (LogvinovLeon) [inline ts/blockchain.ts]: Add return type to this function.

File: ts/blockchain.ts [hunk 3]
```diff
@@ -364,49 +376,48 @@ export class Blockchain {
                 this.stopWatchingExchangeLogFillEventsAsync(); // fire and forget
                 return;
             } else {
-                const args = result.args;
-                const isBlockPending = _.isNull(args.blockNumber);
+                const args = event.args as LogFillContractEventArgs;
+                const isBlockPending = _.isNull(event.blockNumber);
```
Comment (LogvinovLeon) [inline ts/blockchain.ts]: This should never happen, causewe filter only for mined events

Comment (fabioberger) [inline ts/blockchain.ts]: Discussed offline and TODO to verify this created.

**Pull request merged**